### PR TITLE
test(artifact): scylla doesn't use ntp/chrony with systemd-timesyncd

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
As described in https://github.com/scylladb/scylla/issues/8339, if the systemd-timesyncd
service is already installed in the machine, Scylla should not install a different time
synchronization service (ntpd or chronyd) as well. The subtest I added in the artifact
test will make sure of that.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by ShlomiBalalis** [comment]
Test passed on th regular artifact of centos 7, 8, debian 10, ubuntu 18 and ubuntu 20.
Is more testing necessary?

File: artifacts_test.py [hunk 1]
```diff
@@ -200,6 +200,11 @@ def verify_nvme_write_cache(self) -> None:
             write_cache = run(f"cat /sys/block/{nvme_device.removeprefix('/dev/')}/queue/write_cache").stdout.strip()
             self.assertEqual(write_cache, expected_write_cache_value)
 
+    def check_service_existence(self, service_name):
+        res = self.node.remoter.run(f'systemctl list-units --full | grep -Fq "{service_name}"',
+                                    ignore_status=True)
+        return not res.exit_status
```
Comment (rayakurl) [inline artifacts_test.py:206]: In my opinion it's better to return boolian

Comment (ShlomiBalalis) [reply artifacts_test.py:206]: It returns a boolian

File: artifacts_test.py [hunk 2]
```diff
@@ -232,6 +237,24 @@ def test_scylla_service(self):
...
             self.check_scylla()
 
+        with self.subTest("check if scylla unnecessarily installed a time synchronization service"):
+            # Checks https://github.com/scylladb/scylla/issues/8339
+            # If the instance already has systemd-timesyncd
+            is_timesyncd_service_installed = self.check_service_existence(service_name="systemd-timesyncd")
+            is_ntp_service_installed = self.check_service_existence(service_name="ntp")
+            is_chrony_service_installed = self.check_service_existence(service_name="chrony")
+            # Do note: On Redhat based distributions the services are named ntpd and chronyd, while on debian based
+            # distributions they're named ntp and chrony. As such, I looked for the shorter name of the two.
```
[the remark this answers is not in the document]
Comment (fruch) [reply artifacts_test.py:247]: Until someone would improve that function and it won't be using regex.

Maybe it's better to be explicit and check if it's a redhat variant...

**Review by fruch — APPROVED** [review]
LGTM

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
> @fruch Was this backported?

now it is

**Pull request merged**